src/topicmodeller/topicmodeller/topicmodeller.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from boilerpipe.extract import Extractor
+from gensim import corpora, models
+from nltk.corpus import stopwords
+
+import codecs
+import datetime
+import jsonpickle
+import nltk
+import os
+import re
+
+
+class TopicModellerDocument(object):
+    def __init__(self, url, topics):
+        self.url = url
+        self.topics = topics
+
+
+class RepeatableBatchedDocuments(object):
+    def __init__(self, folder, batch_size):
+        self.folder = folder
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        documents = []
+        for file_name in os.listdir(self.folder):
+            (_, document) = _json_sd_file_to_sd_and_tmd(os.path.join(self.folder, file_name))
+            documents.append(document)
+
+            if len(documents) == self.batch_size:
+                yield documents
+                documents = []
+        if len(documents) != 0:
+            yield documents
+
+
+class TopicModeller(object):
+    def __init__(self):
+        self.dictionary = None
+        self.dictionary_words = None
+
+        self.lda = None
+
+    def initialize(self, batched_documents, num_topics):
+        self._initialize_dictionary(batched_documents)
+        self._feed(batched_documents, num_topics)
+
+    def classify(self, document):
+        return self.lda[self.dictionary.doc2bow(document)]
+
+    def load(self, model_data_folder):
+        if os.path.isfile(self._dictionary_file_path(model_data_folder)):
+            self.dictionary = corpora.Dictionary.load(self._dictionary_file_path(model_data_folder))
+            # dictionary.values() is a list, looking into a list is an O(n) operation.
+            # To avoid poor performances on _filter_document method, dictionary words
+            # are "cached" on a set (O(1)).
+            self.dictionary_words = set(self.dictionary.values())
+
+        if os.path.isfile(self._lda_file_path(model_data_folder)):
+            self.lda = models.LdaModel.load(self._lda_file_path(model_data_folder))
+
```

Comment:
if the files are missings, it's a fatal problem for TopicModeller, perhaps throw an exception instead of just let fields to None ?